### DISCUSSION TITLE
Help getting `ngx.ctx` variable

### DISCUSSION THREAD

**Discussion by wallysoncarvalho** [Kong/kong#10880] [Help] [answered]
## HELP with `request-transformer` plugin
<br>

I'm using the [nokia/kong-oidc](https://github.com/nokia/kong-oidc#how-does-it-work) plugin and i noticed that they set the claims of the user as json in a lua variable (in the README and [code](https://github.com/nokia/kong-oidc/blob/master/kong/plugins/oidc/utils.lua#LL83C15-L83C15)). I was expecting to get this in this variable in the `request-transformer` plugin, more especifically inside a **template**.

![image](https://github.com/Kong/kong/assets/19233191/f739be89-9dee-43fc-8840-3b8dbabb4d98)
<br>

I'm using a declarative kong config and i tried this like the one below but it doesnt seem to work:

```
...
   add:
     headers:
       - "x-identifier: $(ngx.ctx.authenticated_consumer.sub)"
...
```

**Does anyone have any ideia what may be going wrong ? Actually I'm not entirely sure that this variable sharing should work.**

**Comment by ms2008** [comment, marked as answer]
According to [this](https://github.com/nokia/kong-oidc/pull/39) `authenticated_consumer` seems to have been changed to `authenticated_credential`, but the documentation doesn't seem to have been updated.

**Comment by wallysoncarvalho** [comment]
thanks @ms2008 , I'll check.
